src/lib/common/sol-bus.c

```diff
@@ -383,9 +383,8 @@ sol_bus_client_free(struct sol_bus_client *client)
     if (!client)
         return;
 
-    destroy_client(client);
-
     sol_ptr_vector_remove(&_ctx.clients, client);
+    destroy_client(client);
```

Comment:
I read somewhere (don't remember where) that even if you don't access the contents, using freed pointers is undefined behavior. A pointer's value after freed is no longer valid.